daml-lf/integration-test-lib/src/main/com/daml/lf/CantonFixture.scala

```diff
@@ -54,18 +55,26 @@ object CantonFixture {
 
 }
 
-trait CantonFixture extends SuiteResource[Vector[Port]] with AkkaBeforeAndAfterAll {
+trait CantonFixtureWithResource[A]
+    extends SuiteResource[(Vector[Port], A)]
+    with AkkaBeforeAndAfterAll {
   self: Suite =>
 
   override protected def afterAll(): Unit = {
     cantonCleanUp()
     super.afterAll()
   }
 
-  final override protected lazy val suiteResource: OwnedResource[ResourceContext, Vector[Port]] = {
+  protected def makeAdditionalResource(ports: Vector[Port]): ResourceOwner[A]
+
+  final override protected lazy val suiteResource
+      : OwnedResource[ResourceContext, (Vector[Port], A)] = {
     implicit val resourceContext: ResourceContext = ResourceContext(system.dispatcher)
-    new OwnedResource[ResourceContext, Vector[Port]](
-      CantonRunner.run(config, cantonTmpDir),
+    new OwnedResource[ResourceContext, (Vector[Port], A)](
+      for {
+        ports <- CantonRunner.run(config, cantonTmpDir)
+        additional <- makeAdditionalResource(ports)
+      } yield (ports, additional),
```

Comment:
What do you think about this @remyhaemmerle-da ?
We have a test that needs additional SuiteResource, so I've refactored this to have a version where you can provide your own resource based on the canton ports

Reply:
LGTM